owltorrent-core/src/main/java/edu/rice/owltorrent/core/serialization/TempParser.java

```diff
@@ -2,19 +2,104 @@
 
 import com.dampcake.bencode.BencodeInputStream;
 import edu.rice.owltorrent.common.entity.Torrent;
+import lombok.NonNull;
+
 import java.io.ByteArrayInputStream;
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
 
+/**
+ * Torrent file parser.
+ *
+ * @author bhaveshshah
+ */
 public class TempParser {
 
-  public Torrent parse(File file) throws IOException {
-    // TODO: revise byte reading method
+  /**
+   * Bencode a Torrent file and extract important attributes
+   *
+   * @param file Torrent file path
+   * @return pruned Torrent object
+   */
+  public static Torrent parse(@NonNull File file) throws IOException {
+    var dict = bencode(file);
+    Torrent torrent = extractAttributes(dict);
+    return torrent;
+  }
+
+  /**
+   * Bencode a Torrent file
+   *
+   * @param file Torrent file
+   * @return dictionary containing all file attributes
+   */
+  public static Map<String, Object> bencode(@NonNull File file) throws IOException {
     ByteArrayInputStream inputStream = new ByteArrayInputStream(Files.readAllBytes(file.toPath()));
     BencodeInputStream bencodeInputStream = new BencodeInputStream(inputStream);
     var dict = bencodeInputStream.readDictionary();
-    System.out.println(dict.keySet());
-    return new Torrent();
+
+    return dict;
+  }
+
+  /**
+   * Extract important attributes to store for later use
+   *
+   * @param dict Containing all file attributes
+   * @return pruned Torrent object
+   */
+  public static Torrent extractAttributes(@NonNull Map<String, Object> dict) {
+    var infoDict = (Map<String, Object>) dict.get("info");
```

Comment:
Consider using static final constants for fields like "info", "announce", etc.

Reply:
Reformatted to use static final constants.